### PR TITLE
[ioctl] Build Execution unittest

### PR DESCRIPTION
**Pull request by LuckyPigeon**
# Description

Build `Execute` unittest for `action` command

refer to #3444 & #3472 

## Type of change
Please delete options that are not relevant.
- [x] Code refactor or improvement

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [x] make test
- [x] TestExecute

**Test Configuration**:
- Firmware version: Windows 10 (WSL Version: Ubuntu 18.04)
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by huangzhiran** [comment]
plz fix the `code smells` and `security hotspots`

**Comment by LuckyPigeon** [comment]
@huangzhiran 
Where do you mean the `code smells`? And the `security hotspots` is about fake account configuration, seems just the naming issue, not an actually `security hotspots`.

**Comment by LuckyPigeon** [comment]
`L251~L253` and `L357~L358` also shouldn't be a `code smell`, it's weird and confused for suddenly shows up an `_failedToGetNonce` constant in test description.

**Comment by LuckyPigeon** [comment]
@huangzhiran Done!

File: test_hint_command.md [hunk 1]
```diff
@@ -0,0 +1 @@
+go test -timeout 30s -run ^TestSendRaw$ github.com/iotexproject/iotex-core/ioctl/newcmd/action
```
Comment (huof6829) [inline test_hint_command.md:1]: Delete temp file for not used in product env.
and directly putting in main dir  is not recommended.

File: ioctl/newcmd/action/action_test.go [hunk 2]
```diff
@@ -270,3 +273,108 @@ func TestSendAction(t *testing.T) {
...
+	apiServiceClient.EXPECT().GetAccount(gomock.Any(), gomock.Any()).Return(accountResponse, nil).Times(5)
+	apiServiceClient.EXPECT().SendAction(gomock.Any(), gomock.Any()).Return(&iotexapi.SendActionResponse{}, nil).Times(2)
+
+	client.EXPECT().SelectTranslation(gomock.Any()).Return("action", config.English).Times(7)
+	client.EXPECT().SetEndpointWithFlag(gomock.Any()).Do(func(_ func(*string, string, string, string)) {
+		// SetEndpointWithFlag take a function as parameter
+	}).AnyTimes()
+	client.EXPECT().SetInsecureWithFlag(gomock.Any()).Do(func(_ func(*bool, string, bool, string)) {
+		// SetInsecureWithFlag take a function as parameter
+	}).AnyTimes()
```
Comment (huof6829) [inline ioctl/newcmd/action/action_test.go:319]: delete `.Do(xxx...)` simply
`Times(7)`

File: test_hint_command.md (hunk 1 above)
Comment (LuckyPigeon) [reply test_hint_command.md]: I didn't notice that I've commited this file, thanks for the advice!

**Review by huof6829 — APPROVED** [review]

**Comment by huof6829** [comment]
@LuckyPigeon  please resolve conflict, thx.

**Comment by LuckyPigeon** [comment]
@huof6829 Done!

**Comment by huof6829** [comment]
> @huof6829

Thx!

**Review by huangzhiran — APPROVED** [review]

**Pull request merged**